src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -4,6 +4,7 @@
   <!--
     Common targets for managed compilers.
   -->
+  <UsingTask TaskName="Microsoft.CodeAnalysis.BuildTasks.MapSourceRoots" AssemblyFile="$(RoslynTargetsPath)\Microsoft.Build.Tasks.CodeAnalysis.dll" Condition="'$(MSBuildAssemblyVersion)' != ''" />
```

Comment:
I looked around for a similar `UsingTask` to import the `Csc` task. I found it in `WriteMSBuildFiles` which makes a `.props` file for the package.
Should we use the same mechanism for the `MapSourceRoots` task?

Reply:
No, this needs to be in the Microsoft.Managed.Core.targets. These are targets are only used via a package reference if you explciitly drop Microsoft.Net.Compilers to your project. By default these are picked up from msbuild, where we insert them to.